### PR TITLE
Fix dockerfiles

### PR DESCRIPTION
**Pull request by arangatang**
*Description of changes:*
Some of the [dockerfiles](examples/dockerfiles) cannot be built/does not execute properly, this PR fixes this. 

These are the problems which this PR addresses:

1. The GPU image crashes on SageMaker due to MXNET_USE_FUSION being enabled. 
2. The GPU image crashes when passing `num_workers` as a hyperparameter. 
3. The mxnet version in the requirements file is unavailable when building the [R image](examples/dockerfiles/Dockerfile.R) so the R image fails to build.

*Problem 1* 

This occurs when the gpu image is run. 
For example running the SimpleFeedForwardEstimator on the electricity dataset with these hyperparameters:
```
epochs	1
forecaster_name	gluonts.model.simple_feedforward.SimpleFeedForwardEstimator
freq	D
prediction_length	7
```

Results in this error (removed redundant parts)

```
AlgorithmError: Traceback (most recent call last): 
File "src/operator/fusion/fused_op.cu", line 647 MXNetError: 
Check failed: compileResult == NVRTC_SUCCESS (5 vs. 0) : NVRTC Compilation failed. 
Please set environment variable MXNET_USE_FUSION to 0.
. . .
```
Following the advice in the logs and setting `MXNET_USE_FUSION=0` fixes the issue.

**Problem 2** (requires problem 1 to be fixed beforehand)
Executing an algorithm with num_workers set to a value causes a CUDA initializtion error. 
A fix for a similar issue was proposed by @lostella for #1054 applying this fix when `num_workers` has been passed fixes problem 2. [link](https://github.com/awslabs/gluon-ts/issues/1054#issuecomment-706161305)

**problem 3**
The latest available mxnet version which can be installed in the R image is mxnet 1.6.0 but the current [mxnet requirement](requirements/requirements-mxnet.txt) is 1.7.0 which breaks the build. Changing the installation requirement to `>=1.6.0` fixes the issue.

I tested the changes on the SimpleFeedForwardEstimator as well as the DeepFactorEstimator on a `ml.g4dn.2xlarge` instance on SageMaker. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/shell/train.py [hunk 1]
```diff
@@ -90,6 +91,8 @@ def run_train(
         if "num_workers" in hyperparameters.keys()
         else None
     )
+    if num_workers:
+        multiprocessing.set_start_method("spawn", force=True)
```
Comment (lostella) [inline src/gluonts/shell/train.py:95]: Maybe minor, but still: this `set_start_method` could be invoked at the beginning of the module, right after the imports, regardless of the number of workers. (If no workers are used, then no subprocess will be launched, so it doesn't matter what start method is being used)

Comment (arangatang) [reply src/gluonts/shell/train.py]: True, I moved it!

**Comment by arangatang** [comment]
It seems as if the setting the multiprocessing method to spawn introduces some additional issues. Maybe this fix only applies to GPUs, ill look into it further.

**Review by mbohlkeschneider — COMMENTED** [review]
Are here any concerns @jaheba, @lostella? Makes sense to me.

**Comment by lostella** [comment]
Looks good to me, let's wait for @arangatang to report on the additional issues he found

**Comment by arangatang** [comment]
```
multiprocessing.set_start_method("spawn", force=True)
``` 
Caused the shell test to fail when a process is forked in the testutils/shell package on [line 139](https://github.com/awslabs/gluon-ts/blob/97ea08cb394f1c0d411234c1b33beff37ae30cd0/src/gluonts/testutil/shell.py#L139). 

Explicitly starting that process as a fork resolves the issue while still allowing gpu acceleration in the shell.

**Tested on**
mxnet:
mxnet-cu101 1.7.0
mxnet 1.7.0

GPU:
Titan RTX

NVCC:
version 10.1

**Comment by arangatang** [comment]
I recon the changes are good to go! 
What do you think?

@lostella @mbohlkeschneider

**Comment by arangatang** [comment]
I believe the failing CodeQL test is unrelated to my changes, should I merge it?

**Comment by jaheba** [comment]
Yes, let's merge it.

**Pull request merged**